### ISSUE TITLE
How to fit column width to cell text length?

### ISSUE THREAD

**Issue by zapjonny** [bannzai/SpreadsheetView#23]
Thanks for a good looking spreadsheetview, I am interested in the timetable in particular.

I have two questions, I looked at the code but didn't find an immediate way to do the below so I'll just ask:

1. Column widths are specified by the delegate - can this be handled by autolayout instead, someway?

2. How would I turn off the titles (channel names)? I'd like to not show that top row at all.

**Comment by kishikawakatsumi** [comment]
@zapjonny 

> Column widths are specified by the delegate - can this be handled by autolayout instead, someway?

What you want is `UITableViewAutomaticDimension` and` estimatedHeight` in `UITableView`? Currently, it is not supported. This is because the entire width and height are required before laying out individual cells. (It is the same reason that `UITableView` requires` estimatedHeight`.) If you want to calculate the height using AutoLayout, you need to do it yourself.

> How would I turn off the titles (channel names)? I'd like to not show that top row at all.

Configure the data source to achieve the results you want. Probably make `frozenRow` returning zero (or just remove the method, default is zero) then delete the header data.

**Comment by zapjonny** [comment]
OK, think I got those things...

I implemented, or tried, to implement something similar but didn't get very far. So this is definitely a big help. However I'd probably need two more things, and not sure they are implemented or how hard it would be to add them:

1. <s>Something similar to the scrollViewDidScroll protocol function of UIScrollView. I need to update a fixed label with what's currently showing on screen. Maybe I could just use a timer for that, if all else fails. Looks like I could use visibleCells() for that. I guess SpreadsheetView does not extend a UIScrollView, like a UITableView does.</s>
Nevermind. I could do it by adding this to my code:

```
extension SpreadsheetView {
	func scrollViewDidScroll(_ scrollView: UIScrollView) {
		print("SpreadsheetView scrolled!")
	}
}
```

Swift makes things easy sometimes 👍 

Update again: Ok I guess I accidentally overrode some actual code with that. Seems I would be better off adding that scrollViewdidScroll() to SpreadsheetViewDelegate after all.


2. scroll to a certain cell programmatically.


Once again thank you for your effort. This is probably the best spreadsheet/timetable out there.
